xarray/core/nputils.py

```diff
@@ -206,6 +206,11 @@ def f(values, axis=None, **kwargs):
                 # to ddof=1 above.
                 if pycompat.mod_version("numbagg") < Version("0.7.0"):
                     kwargs.pop("ddof", None)
+                if name == "nanquantile":
+                    if kwargs.get("method", None) != "linear":
+                        return
```

Comment:
If name is `nanquantile`, but method is not `linear`, I think this is going to return `None`. But instead we should let the rest of the function execute.

So could we instead merge L209 & L210, into something like:

```suggestion
                if name == "nanquantile" and kwargs.get("method", "linear") == "linear":
```

Reply:
https://github.com/pydata/xarray/pull/8684/commits/857002d007a8d519f40eeccf0be5ce7d8d71197c Thanks for the pointer!

Reply:
Some tests are still failing though. Might take some days until I have some spare time to look further into this.